File: exam-slammr-webapp/src/test/java/uk/co/claritysoftware/exam/slammr/webapp/web/model/question/FurtherReadingTest.java
package uk.co.claritysoftware.exam.slammr.webapp.web.model.question;

import nl.jqno.equalsverifier.EqualsVerifier;
import nl.jqno.equalsverifier.Warning;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * Unit test class for {@link FurtherReading}
 */
public class FurtherReadingTest {

    @Test
    public void shouldGenerateEmptyFurtherReading() {
        // Given
        FurtherReading expectedFurtherReading = FurtherReading.builder()
                .description(null)
                .referenceLocation(null)
                .build();

        // When
        FurtherReading furtherReading = FurtherReading.generateEmptyFurtherReading();

        // Then
        assertThat(furtherReading).isEqualTo(expectedFurtherReading);
    }

    @Test
    public void shouldHonourEqualsHashcodeContract() {
        EqualsVerifier.forClass(FurtherReading.class)
                .suppress(Warning.NONFINAL_FIELDS)
                .verify();
    }

}

File: exam-slammr-webapp/src/test/kotlin/uk/co/claritysoftware/exam/slammr/webapp/web/model/question/FurtherReadingTest.kt
package uk.co.claritysoftware.exam.slammr.webapp.web.model.question

import nl.jqno.equalsverifier.EqualsVerifier
import nl.jqno.equalsverifier.Warning
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

/**
 * Unit test class for [FurtherReading]
 */
class FurtherReadingTest {

	@Test
	fun shouldGenerateEmptyFurtherReading() {
		// Given
		val expectedFurtherReading = FurtherReading.builder()
				.description(null)
				.referenceLocation(null)
				.build()

		// When
		val furtherReading = FurtherReading.generateEmptyFurtherReading()

		// Then
		assertThat(furtherReading).isEqualTo(expectedFurtherReading)
	}

	@Test
	fun shouldHonourEqualsHashcodeContract() {
		EqualsVerifier.forClass(FurtherReading::class.java)
				.suppress(Warning.NONFINAL_FIELDS)
				.verify()
	}

}